client/components/error-boundary/index.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * WordPress dependencies
+ */
+import { Component } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { Notice } from '@wordpress/components';
+
+class ErrorBoundary extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.state = {
+			error: null,
+		};
+	}
+
+	static getDerivedStateFromError( error ) {
+		return { error };
+	}
+
+	componentDidCatch( error, info ) {
+		if ( this.props.onError ) {
+			this.props.onError( error, info );
+		}
+	}
+
+	render() {
+		if ( ! this.state.error ) {
+			return this.props.children;
+		}
+
+		return (
+			<Notice status="error" isDismissible={ false }>
+				{ __(
+					'There was an error rendering the component, please try again. Contact support for assistance if the problem persists.',
+					'woocommerce-payments'
+				) }
+				<br />
+				{ this.state.error.toString() }
```

Comment:
We could probably update the error message to make it more useful for users. Glad to adjust it.